frontend/src/pages/projects/screens/detail/connections/__tests__/ConnectionsTable.spec.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import ConnectionsTable from '~/pages/projects/screens/detail/connections/ConnectionsTable';
+import { Connection } from '~/pages/projects/screens/detail/connections/types';
+
+const connections: Connection[] = [
+  {
+    kind: 'Secret',
+    apiVersion: 'v1',
+    metadata: {
+      name: 'connection1',
+      namespace: 'ds-project',
+      labels: { 'opendatahub.io/dashboard': 'true', 'opendatahub.io/managed': 'true' },
+      annotations: {
+        'opendatahub.io/connection-type': 's3',
+        'openshift.io/display-name': 'connection1',
+        'openshift.io/description': 'desc1',
+      },
+    },
+    data: {},
+  },
+];
```

Comment:
The old type of `DataConnection` doesn't fully line up. I can relax some of the properties in `Connection` and make them nullish which helps. But then the `data` property is a pain to properly convert, since it's an enum of the `AwsKeys`.

I could just do the following (but the description annotation doesn't exist in `DataConnection`):
```
export const mockConnection = (args: MockDataConnectionType): Connection =>
  mockDataConnection(args).data as unknown as Connection;
```
(And yes the `.data` is needed because that is actually the `SecretKind`)

Reply:
Let's make a new mock util for these connection and then we can eventually remove the old data connection mock and types.